### DISCUSSION TITLE
Система валюты на сервере

### DISCUSSION THREAD

**Discussion by Yorodoshi** [TerraFirmaGreg-Team/TFG-Modpack-1.20.x#174] [Ideas]
### Версия используемого модпака

0.1.2

### Связанная проблема

Нам нужна торговая капиталистическая система с продвинутой централизованной валютой.

### Ваше решение

Решение:

Необходимые правила:

1. Бан за твинки.
2. Невозможность перевода без двустороннего подтверждения транзакции. А так-же все следствия этого правила.
3. Общий капитал команд. Капитал сохраняется за командой, а не за её членами, т.е. косвенно принадлежит создателю. Право на управление капиталом команды идёт так: 
Owner, Promote перевод капитала и получение переводов.
Member может только получать переводы капитала.
Люди без команды имеют только личный счёт, при вхождении в команду, весь их счёт переводится в счёт команды, после выхода этого члена капитал остаётся у команды.
4. Общие приваты команд по правилу

Про приваты: множитель не может уменьшить кол-во приватов в тиме.

Решение:
Система рангов при капитализме:

1. Рабочий - Является обыкновенным гражданином, ведущий простой потребительский образ жизни не ведущий бизнесов.
Требуемый капитал: 0-200 монет
Множитель привата X1

2. Рыночный агент - Новичок в мире капитализма, только начинающий осваивать торговлю и умение вести успешные сделки.
Требуемый капитал: 200-500 монет
Множитель привата X2/membersNum

3. Капиталистический горняк - Умелый добытчик ресурсов, способный выгодно торговать рудами.
Требуемый капитал 500-1 500 монет
Множитель привата X2,5/membersNum
4. Феррум-трейдер - Эксперт по торговле сплавами, способный проводить успешные сделки в сфере продвинутой металлургии.
Требуемый капитал 1 500-4 000 монет
Множитель привата X3/membersNum

5. Технобизнесмен - Опытный предприниматель в сфере технологий, создающий и продавая продвинутые устройства и машины.
Требуемый капитал 4 000-10 000 монет
Множитель привата X5/membersNum

6. Электро-магнат - Предприниматель с большим опытом в торговле почти всем связанным с электричеством, турбинами, целостными системами.
Требуемый капитал 10 000-30 000 монет
Множитель привата X7/membersNum

7. Химический магнат - Магнат, управляющий химическими предприятиями, производящими крайне продвинутые материалы и ресурсы.
Требуемый капитал 30 000 - 70 000 монет
Множитель привата X10/membersNum

8. Архитектор торговых центров - Строитель крупных коммерческих комплексов и рыночных площадей, создающий места для успешных сделок.
Требуемый капитал 70 000 - 190 000 монет
Множитель привата X15/membersNum

9. Финансовый стратег - Умелый инвестор, осуществляющий успешные финансовые операции и управляющий портфелем высокотехнологичных активов.
Требуемый капитал 190 000 - 500 000 монет
Множитель привата X20/membersNum

10. Космо-корпоратист - Владелец космических предприятий, осуществляющий торговлю в космическом пространстве.
Требуемый капитал 500 000 - 1 300 000 монет
Множитель привата X40/membersNum

11. Технократ-магнат - Магнат, контролирующий важнейшие технологические отрасли и диктующий условия на рынке.
Требуемый капитал 1 300 000 монет+
Множитель привата X50/membersNum

Каким способом будут производиться монеты и по какой формуле ?

Вводные данные: 
R(ранг, int), x(преодолено ед.изм[м]), N(количество), d(изменение за некое время t), L(время проведенное в игре ед.изм[мин])

1. Преодолено в полёте:
coins = (dx×0.0005)×(R^0.50) \\ Пример: рабочий при пролёте 1км получает 0,5 монет, Технобизнесмен 1,11 монету

2. Регенерация ХП(За расчёт берём ТФК единицы):
coins = (dN×0.003)×(R^1,9) \\Пример: Капиталистический горняк зарегенил 100хп и получил 2,4 монет, Космо-корпоратист получит уже 79,43 монеты

3. Зарплата:
Особенная величина необходимая для прироста покупательной способности классов с низким капиталом
coins = (1/R)×2×L
 
5. Преодолено бегом:
coins = (dx×0.001)×(R^0.6) 

6. Преодолено пешком:
coins = (dx×0.001)×(R^1.4)

7. Прыжки:
coins = (dN×0.003)×(R^1.085)

8. Поспать на кровати:
coins = (dN×1)×(R^2)

9. Смерти:
coins = -((dN×0,5)×(2^(1.5^R))) \\Пример: Рабочий при смерти тратит 1,41 монеты, а Технократ-магнат 16 384 монеты
           *В случае смерти от игрока убийца получает 0.5 от числа coins

###Что это даст ?

Заставит игроков общаться с друг другом ещё больше, что ещё больше подогреет их интерес к игре и добавит удовольствия.

В будущем с появлением поездов Create сделает возможным деление производства материалов что способствует большему развитию.

Научит ценить игроков их жизнь. В последнее время люди очень часто умирают, не делают броню, предпочитают суицид еде и прочее.

### Дополнительная информация

С приватами не знаю что по человечески делать, додумывайте

Мне сейчас нужен коммент для того чтобы я понял что вообще вы (админы) можете сделать. Какие есть инструменты ? Можно ли сделать все что я написал в необходимых правилах?

**Comment by FriendlyCube** [comment]
0.o

**Comment by Andrei-Mudrov-archmage** [comment]
Mi za kommunizm

**Comment by Andrei-Mudrov-archmage** [comment]
Davayte uberyom privat i sdelaem anarhicheskiy server?
